### PR TITLE
build: allow building on Windows

### PR DESCRIPTION
**Pull request by compnerd**
This allows us to build on Windows, though not entirely.  The assembly
routines need to be generated for the Windows targets, but this is
required to get to that point.  With this set of changes, we are able to
get to the point where we can start linking the `crypto-shasum` target,
which exposes the missing symbols.

_[One line description of your change]_

### Checklist
- [x] I've run tests to see all new and existing tests pass
- [x] I've followed the code style of the rest of the project
- [x] I've read the [Contribution Guidelines](CONTRIBUTING.md)
- [x] I've updated the documentation if necessary

#### If you've made changes to `gyb` files
- [x] I've run `.script/generate_boilerplate_files_with_gyb` and included updated generated files in a commit of this pull request

### Motivation:

_[Explain here the context, and why you're making that change. What is the problem you're trying to solve.]_

### Modifications:

_[Describe the modifications you've done.]_

### Result:

_[After your change, what will change.]_

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @Lukasa @tomerd

**Review by Lukasa — APPROVED** [review]

**Comment by Lukasa** [comment]
@swift-server-bot add to whitelist

**Comment by Lukasa** [comment]
@swift-server-bot test this please

File: Package.swift [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:5.1
+// swift-tools-version:5.3
```
Comment (Lukasa) [inline Package.swift:1]: I'm afraid we can't bump the tools version: swift-crypto supports back to 5.1. We _can_ add a new Package.swift if we have to, though it complicates matters a bit.

Is there any particular reason not to just define WIN32_LEAN_AND_MEAN in all cases?

Comment (compnerd) [reply Package.swift:1]: The best reason that I have against that is hygiene and the reminder that it is in the user's namespace (it is not underscore prefixed, so this symbol is perfectly cromulent to be used by a user in their macOS program).  I can just unconditionally define it.  We could also sprinkle it amongst the various source files as necessary under a guard.

Comment (Lukasa) [reply Package.swift:1]: I think I'm happy to unconditionally define it for this target. cSettings are totally available in the 5.1 tools version, we just can't use _platform-specific_ settings.

Comment (compnerd) [reply Package.swift]: I can agree to that and dislike it right? :-p.  Done.

**Review by Lukasa — APPROVED** [review]

**Pull request merged**